### PR TITLE
[LMS-76] Automatic publish for Google Classroom

### PR DESCRIPTION
**Pull request by bradbanister**

### CODE REVIEW

**Comment by bradbanister** [comment]
A comment

**Comment by bradbanister** [comment]
This works when triggered by an in-branch push.  See: https://github.com/Ed-Fi-Exchange-OSS/LMS-Toolkit/runs/2018651912

Changed to a manual trigger (workflow_dispatch event), but needs to be merged into main for the shiny green button to be visible.  Please review for all but that, and post merge we can do a final test of the manual trigger.

One question is if there is a known better way to share between this publish workflow file and the test workflow file.  I couldn't find an approach that would work.  Multiple jobs in a file didn't seem to fit the bill, and as far as I can tell one workflow triggering another lacks clear support.

File: .github/workflows/publish-google-classroom.yml
Comment (stephenfuqua) [inline .github/workflows/publish-google-classroom.yml:44]: I agree that we need to re-run some quality checks before publishing, since it is possible to merge something and ignore an error in the trigger on push to `main`. However, perhaps we can simplify a bit by just running `pytest` without capture code coverage? Running type check also helps avoid production bugs, but I wonder if running the linter is really necessary?

File: eng/build.py
Comment (stephenfuqua) [inline eng/build.py:170]: 👍

**Review by stephenfuqua — APPROVED** [review]
Approved, but not merging pending @bradbanister's response to a question about streamlining.

**Comment by stephenfuqua** [comment]
> One question is if there is a known better way to share between this publish workflow file and the test workflow file. I couldn't find an approach that would work. Multiple jobs in a file didn't seem to fit the bill, and as far as I can tell one workflow triggering another lacks clear support.

Not that I've been able to find... although we might be able to create a custom Action that is a composite of other things. Hadn't thought about that before. I have commented in a GitHub survey that this is the top missing feature in my view.

File: .github/workflows/publish-google-classroom.yml
Comment (bradbanister) [reply .github/workflows/publish-google-classroom.yml:44]: That makes sense.  I'll update.

**Review by stephenfuqua — APPROVED** [review]

**Pull request merged**
